user_guide_src/source/concepts/factories.rst

```diff
@@ -0,0 +1,117 @@
+files and classes within ``Models``, so you can request a model by its shorthand base name::
+
+	use CodeIgniter\Config\Factories;
+
+	$users = Factories::models('UserModel');
+
+Or you could also request a specific class::
+
+	$widgets = Factories::models('Some\Namespace\Models\WidgetModel');
+
+Next time you ask for the same class anywhere in your code, ``Factories`` will be sure
+you get back the instance as before::
+
+	class SomeOtherClass
+	{
+		$widgets = Factories::models('WidgetModel');
+
+
+Factories Behavior
+==================
+
+The default behavior of ``Factories`` might not always work for ever component. For example,
+say your component name and its path do not align, or you need to be sure that only a certain
+type of class instances are returned, or your component classes require additional parameters
+for their constructor. Learn how to use ``Factories`` and its configuration to meet your needs.
+
+Factory Parameters
+------------------
+
+By default, ``Factories`` assumes that you want to locate a shared instance of a component
+with the provided name. The convenience of ``Factories`` for locating classes is sometimes
+desirable without using its shared instance storage. By adding a second parameter to the
+magic static call, you can control whether ``Factories`` will return a new or shared instance
+each time::
+
+	$users = Factories::models('UserModel', true); // Default; will always be the same instance
+	$other = Factories::models('UserModel', false); // Will always create a new instance
+
+Additionally, any more parameters passed at the same time will be forwarded on to the class
+constructor, making it easy to configure your shared instance on-the-fly. For example, say
+your app uses a separate database for authentication and you want to be sure that any attempts
+to access user records always go through that connection::
+
+	$conn  = db_connect('AuthDatabase');
+	$users = Factories::models('UserModel', true, $conn);
+
+Now any time the ``UserModel`` is loaded from ``Factories`` it will in fact be returning the
+shared instance that uses the alternate database connection.
+
+Factory Configuration
+---------------------
+
+What works for one component may not work for all. ``Factories`` supports alternate behavior
+at the component level via configurations. A configuration consists of any of the settings,
+which will fall back on the default if not provided.
+
+======================= ============== ==================================================================================================================== ===================================================
+Key                     Type           Description                                                                                     Default
+======================= ============== ==================================================================================================================== ===================================================
```

Comment:
Python syntax check screams the table is malformed. I think this is because the dividers for `Description` and `Default` are continuous instead of broken. Also, I think the upper and bottom dividers are not of same length.